_posts/2019-02-14-Viewability.md

```diff
@@ -0,0 +1,115 @@
+For the measurement model we take all impressions and use the measurable column as target in a supervised classification model (namely a Neural Network).
+The output of this model is the probability of each impression to be measurable. The viewability model takes all impressions, which are measurable and uses the viewability column as target.
+This model gives a conditional probability of viewability for impressions which are measurable. Thus, the total probability of viewability is:
+
+TODO: Insert nice looking formula here
+
+We perform a training of this model every couple of hours to be sure to make use of all recent campaign and publisher data.
+
+#### Application
+
+We can use the combination of the two models in our real-time bidding platform to get the probability for each impression we want to bid on. Then, we can scale the bidding price with this probability.
+So, if a viewable impression has the value of one cent to us and the probability of a view is 50%, we are bidding half a cent. If we buy two impressions with that probability for that price, we will get one viewable impression on average and pay one cent.
+Mission accomplished. There are two interesting aspects to this: First, we do not buy the viewable impressions for the cheapest possible price. This is approached with another mechanism. We are buying the viewable impression for a defined price and can allways be sure, to buy for that price or lower.
+This allows us to sell the viewable impression for a specific price. If we configure a price, which is too low we might not be able to buy the desired amount of impressions, but we will not over-pay.
+And second, this mechanism does not necessarily improve the rate of viewable impressions. We do not care if we buy many impressions with a low viewability rate or a few with a high. Since the potential customer only cares about viewable impressions, this is OK.
+
+#### Monitoring
+
+But this whole idea only works, if we have two classification models which give us an unbiased probability. Thus, if the output of the model is 0.5, half of the impressions should lead to a view. A lof of data-science libraries support this kind of output but many aspects of the data and the model can break this concept.
+Therefore, we monitor this capability by regularly producing and checking the following plot:
+<p align="center"> <img src="../images/diag.png"> </p>
+
+On the x-axis you can see the output of the machine-learning model. On the y-axis is the ratio of events (fraction of measures/views) in a bin of the output. If the output of the model is a probability, the data points should all lie on the diagonal y = x within their uncertainty.
+The uncertainty originates from the inherent error on the fraction due to the probabalistic nature of the numbers.
```

Comment:
suggest 
"This uncertainty originates from probabalistic nature of the underlying numbers." (there is no error)